src/helpers/getByAPI.js

```diff
@@ -20,12 +20,29 @@ const getNodeByText = (node, text) => {
       const textChildren = getChildrenAsText(node.props.children, Text);
       if (textChildren) {
         const textToTest = textChildren.join('');
-        return typeof text === 'string'
-          ? text === textToTest
-          : text.test(textToTest);
+        if (typeof text === 'string') {
+          if (text === textToTest) {
+            return { exact: true };
+          } else {
+            if (textToTest.includes(text)) {
```

Comment:
Using includes here posed a lot of problem since it means that the parent nodes will also match. Hence the _significant_ update to the logic.